lib/mix/lib/mix/dep.ex

```diff
@@ -103,6 +103,29 @@ defmodule Mix.Dep do
     end
   end
 
+  @doc """
+  Clears loaded dependencies from the cache for the current environment.
+  """
+  def clear_cached() do
+    project = Mix.Project.get()
+    key = {:cached_deps, Mix.env(), project}
+    cache = project && Mix.ProjectStack.read_cache(key)
+
+    if cache do
+      Enum.map(cache, fn dep ->
+        case Mix.ProjectStack.read_cache({:app, dep.app}) do
```

Comment:
What is the module being purged in this case? The mix.exs module? We are still going to have side-effects here if other modules are defined in the same file. Is purging them really necessary? Isn't the concern exclusively about rebar?

Reply:
What nerves needs to do is to load some dependencies, set up the environment with the help of those deps, and then load the rest of the deps. In this case the problem was that the rebar config was loaded and cached before setting up the environment but we can have the same problem with mix deps so I am doing a more general solution.

Reply:
We are unloading the mix.exs modules. I'm not sure what to do about other modules in the file.

Reply:
Yeah, I am sure this will cause at least warnings, so it is a no-no from my side. It is most likely that the dynamic part of a mix.exs is inside `def project`, so we don't need to remove the whole module, we just need to invoke the project function again.

Reply:
I have just verified that we always call `project`, so I would recommend just getting rid of this Enum.map.